design/one-pager-resource-connectivity-mvp.md

```diff
@@ -0,0 +1,653 @@
+    - mycoolapp
+    tier: db-n1-standard-1
+    backupConfiguration:
+      startTime: 2019-01-01 00:00:00
+      location: us-west1
+      enabled: true
+      binaryLogEnabled: true
+      replicationLogArchivingEnabled: true
+    pricingPlan: PER_USE
+    replicationType: ASYNCHRONOUS
+    activationPolicy: ON_DEMAND
+    ipConfiguration:
+      ipv4Enabled: true
+      # authorizedNetworks whitelists the specified _public_ CIDRs.
+      authorizedNetworks:
+      - name: mycoolcidr
+        value: 73.98.0.0/16
+        expirationTime: datetime
+      requireSsl: false
+      # privateNetwork whitelists any instance in the specified GCP VPC network.
+      privateNetwork: /projects/mycoolproject/global/networks/mycoolvpc
+    locationPreference:
+      followGaeApplication: mycoolapp
+      zone: us-central1-a
+    databaseFlags:
+    - name: cool
+      value: very
+    databaseReplicationEnabled: true
+    crashSafeReplicationEnabled: true
+    dataDiskSizeGb: 50
+    dataDiskType: PD_SSD
+    maintenanceWindow:
+      hour: 23
+      day: 2
+      updateTrack: stable
+    storageAutoResize: true
+    storageAutoResizeLimit: 500
+    userLabels:
+      cool: very
+  masterInstanceName: mycoolmastername
+  failoverReplica:
+    name: mycoolreplicaname
+  replicaConfiguration:
+    mysqlReplicaConfiguration:
+      dumpFilePath: dump
+      username: cooluser
+      password: secretpassword
+      connectRetryInterval: 60
+      masterHeartbeatPeriod: 100
+      caCertificate: PEMPEMPEM
+      clientCertificate: PEMPEMPEM
+      clientKey: PEMPEMPEM
+      sslCipher: supersecure
+      verifyServerCertificate: true
+    failoverTarget: false
+```
+
+#### Google Kubernetes Engine
+
+The Crossplane `GKECluster` managed resource is particularly misleading. Its API
+definition has a fairly comprehensive (albeit confusingly flattened) facsimile
+of the associated GKE API object, but only a small subset of these fields are
+actually parsed and submitted to the GKE API. Only the fields that are parsed
+and submitted are included in the below example:
+
+```yaml
+---
+apiVersion: compute.gcp.crossplane.io/v1alpha1
+kind: GKECluster
+metadata:
+  namespace: crossplane-system
+  name: example
+spec:
+  nameFormat: mycoolname
+  clusterVersion: 1.13
+  createSubnetwork: false
+  enableIPAlias: true
+  labels:
+    cool: very
+  machineType: n1-standard-2
+  clusterIPV4CIDR: 192.168.0.0/16
+  nodeIPV4CIDR: 10.0.0.0/8
+  serviceIPV4CIDR: 172.16.0.0/24
+  nodeLabels:
+  - coollabel
+  numNodes: 6
+  zone: us-central1-a
+  Scopes:
+  - mycoolnodescope
+  # This setting is not currently read from the GKE resource class.
+  username: cooluser
+```
+
+By comparison, a direct translation of the [GKE cluster external resource]'s
+writable API object fields to a Kubernetes YAML specification would be as
+follows. Note that the GKE API contains several deprecated fields, all of which
+are superceded by others (e.g. `nodeConfig` is superceded by `nodePools`). The
```

Comment:
IFF `nodePools` are provided. There is a subtle and yet very important difference. `nodePools` have a lifecycle of their own and probably should be modeled as such all-together.

Reply:
See our earlier discussion around `CloudSQLInstance` fields. I was torn here as to whether I should omit the duplicative, deprecated fields, and omitted them mostly for brevity's sake. It's my opinion that we should probably include them all, and let the managed resource author determine which ones make sense to set (or not) per the upstream API documentation. i.e. Recreate the upstream API, don't try to be opinionated about it or make it easier (at the managed resource level).